crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -30,15 +30,21 @@ impl PgManager {
 
     /// Create a new underlying reader, which is used by this type as well as other data providers.
     pub(crate) fn reader(db_url: impl Into<String>) -> Result<IndexerReader, Error> {
-        Self::reader_with_config(db_url, DEFAULT_SERVER_DB_POOL_SIZE)
+        Self::reader_with_config(
+            db_url,
+            DEFAULT_SERVER_DB_POOL_SIZE,
+            DEFAULT_REQUEST_TIMEOUT_MS,
+        )
     }
 
     pub(crate) fn reader_with_config(
         db_url: impl Into<String>,
         pool_size: u32,
+        timeout_ms: u64,
     ) -> Result<IndexerReader, Error> {
         let mut config = PgConnectionPoolConfig::default();
         config.set_pool_size(pool_size);
+        config.set_connection_timeout(Duration::from_millis(timeout_ms));
```

Comment:
the connection timeout is actually "connection establishing wait-for timeout" per this comment
```
    /// Calls to `Pool::get` will wait this long for a connection to become
    /// available before returning an error.
```
`statement_timeout` for long-running query? `idle_timeout` is for clean-up, we do not have it in config, and its default is 10 min

Reply:
Ah thanks, I will add `statement_timeout` as well, thanks for flagging.

Reply:
Thanks for catching this, i had forgotten the difference myself :D